### PR TITLE
Option invert map movement

### LINKED ISSUE #4662: Option to invert map movement
https://www.widelands.org/forum/topic/5144
Add an option to invert the map movement direction when clicking+dragging with the right mouse button.

### PR DESCRIPTION
**Pull request by greatmomo**
Added menu option to invert mouse movement via click and drag. 

This is my first time contributing to an open source project, so don't be afraid to give feedback or constructive criticism. Thanks.

### CODE REVIEW

**Comment by greatmomo** [comment]
I'm not sure how to link my pull request to the Issue: Option to invert map movement #4662

File: src/wui/mapview.cc [hunk 1]
```diff
@@ -324,6 +324,8 @@ MapView::MapView(
      dragging_(false),
      edge_scrolling_(parent && !parent->get_parent() /* not in watch windows */ &&
                      get_config_bool("edge_scrolling", false)),
+     invert_movement_(parent && !parent->get_parent() /* not in watch windows */ &&
+                     get_config_bool("invert_movement", false)),
```
Comment (Noordfrees) [inline src/wui/mapview.cc]: This feature should be used for watch windows as well for consistency:
```suggestion
     invert_movement_(get_config_bool("invert_movement", false)),
```

File: src/ui_fsmenu/options.cc [hunk 2]
```diff
@@ -301,6 +301,10 @@ Options::Options(MainMenu& fsmm, OptionsCtrl::OptionsStruct opt)
                      UI::PanelStyle::kFsMenu,
                      Vector2i::zero(),
                      _("Scroll when the mouse cursor is near the screen edge")),
+     invert_movement_(&box_ingame_,
+                     UI::PanelStyle::kFsMenu,
+                     Vector2i::zero(),
+                     _("Invert mouse movement via right click")),
```
Comment (Noordfrees) [inline src/ui_fsmenu/options.cc]: Sounds a bit unclear IMHO. How about something like
```suggestion
                     _("Invert click-and-drag map movement direction")),
```

**Review by Noordfrees — COMMENTED** [review]
Hi and thanks for your patch :) 

Code looking good, two comments inline.

Since Widelands is currently in pre-release feature freeze merging this will have to wait until after the 1.0 release.

**Comment by greatmomo** [comment]
Thanks for the feedback! I committed your suggestions.

**Review by Noordfrees — APPROVED** [review]
Code LGTM, tested and working well :)

**Pull request merged**